### PR TITLE
Added typing to even more files 

### PR DESCRIPTION
**Pull request by abidlabs**
* `examples.py`
* `exceptions.py`
* `external.py`
* `external_utils.py`
* `serializing.py`
* `layouts.py`
* `flagging.py`

### CODE REVIEW

File: gradio/components.py [hunk 1]
```diff
@@ -2353,6 +2353,7 @@ def serialize(
         self, x: str, load_dir: str = "", encryption_key: bytes | None = None
     ) -> Dict:
         serialized = FileSerializable.serialize(self, x, load_dir, encryption_key)
+        assert serialized is not None
```
Comment (freddyaboulton) [inline gradio/components.py:2356]: Is this to make the type checker happy? Can we make a change to `serialize` so that we don't have to add these asserts?

Comment (abidlabs) [reply gradio/components.py:2356]: Hmmmm I think so. Let me check

Comment (abidlabs) [reply gradio/components.py]: Actually `serialized` _can_ be `None` if the input `x` is `None`. Handled that case now

File: gradio/external_utils.py [hunk 2]
```diff
@@ -41,7 +46,7 @@ def get_tabular_examples(model_name: str) -> Dict[str, List[float]]:
     # replace nan with string NaN for inference API
     for data in example_data.values():
         for i, val in enumerate(data):
-            if isinstance(val, numbers.Number) and math.isnan(val):
+            if isinstance(val, float) and math.isnan(val):
```
Comment (aliabid94) [inline gradio/external_utils.py:49]: not possible for this to be int?

File: gradio/flagging.py [hunk 3]
```diff
@@ -51,9 +52,9 @@ def _get_dataset_features_info(is_new, components):
                 headers.append(component.label + " file")
                 for _component, _type in file_preview_types.items():
                     if isinstance(component, _component):
-                        infos["flagged"]["features"][component.label + " file"] = {
-                            "_type": _type
-                        }
+                        infos["flagged"]["features"][
+                            component.label or "" + " file"
```
Comment (aliabid94) [inline gradio/flagging.py:56]: `(component.label or "") + " file"` 
not sure if order of operations does that already but better to be explicit

**Review by aliabid94 — APPROVED** [review]
left minor comments, lgtm otherwise

File: gradio/flagging.py (hunk 3 above)
Comment (abidlabs) [reply gradio/flagging.py:56]: Sg thanks

File: gradio/external_utils.py (hunk 2 above)
Comment (abidlabs) [reply gradio/external_utils.py:49]: It is but an int would never be nan so the second part of the condition would fail

**Pull request merged**